src/application/application.ts

```diff
@@ -81,6 +82,12 @@ class Application {
     }
 
     private async startService(id: string) {
+        // service is already starting.
+        if (this.serviceIDs.indexOf(id) >= 0) {
+            return
+        }
+        this.serviceIDs = [...this.serviceIDs, id];
```

Comment:
@antho1404 do you think we should rm the service id from the array when we have an error returned while starting the service? because caller may want to retry starting the service.

Reply:
We can improve this part later I think, I'll add an new issue